[Instrumentation.AspNetCore] Fix issue causing Activity is not exported when used with custom propagator/sampler. (#4637)
open-telemetry/opentelemetry-dotnet: src/OpenTelemetry.Instrumentation.AspNetCore/CHANGELOG.md

  which attributes are emitted by setting the environment variable
  `OTEL_SEMCONV_STABILITY_OPT_IN`.

* Fixed an issue affecting NET 7.0+. If custom propagation is being used

>   and a custom sampler adds tags to an Activity that Activity would be dropped.

Reviewer: ```suggestion
  and tags are added to an Activity during sampling then that Activity would be dropped.
```
